addons/pos_online_payment/controllers/payment_portal.py

```diff
@@ -96,7 +96,7 @@ def pos_order_pay(self, pos_order_id, access_token=None, exit_route=None):
 
         user_sudo = request.env.user
         if not pos_order_sudo.partner_id:
-            user_sudo = request.env.ref('base.public_user')
+            user_sudo = request.env.ref('base.public_user').sudo()
```

Comment:
maybe we need to use `_get_public_user` from res.company to get the proper public user.

Reply:
Oh nice, but that method is in website module, therefore i move it into base module
Please review